src/ops/tag.rs

```diff
@@ -128,8 +147,14 @@ impl OpTag {
 
     /// Returns whether the set is empty.
     #[inline]
-    pub fn is_empty(&self) -> bool {
-        self == &OpTag::None
+    pub const fn is_empty(&self) -> bool {
+        matches!(self, &OpTag::None)
+    }
+
+    /// Constant equality check.
+    #[inline]
+    pub const fn eq(self, other: OpTag) -> bool {
```

Comment:
Unfortunately `==` is not a constant operation in general (because it is a trait method that can be implemented using non-constant things).
We can only do that with integers and other "structurally equivalent" primitive types that the compiler knows about.

Enums without data should be in that category, but rustc doesn't support that yet.

Hence the cast to use the constant int equality check.